bundles/org.openhab.binding.radiobrowser/src/main/java/org/openhab/binding/radiobrowser/internal/api/RadioBrowserApi.java

```diff
@@ -0,0 +1,359 @@
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.radiobrowser.internal.api;
+
+import static org.openhab.binding.radiobrowser.internal.RadioBrowserBindingConstants.*;
+
+import java.io.UnsupportedEncodingException;
+import java.net.InetAddress;
+import java.net.URLEncoder;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpMethod;
+import org.openhab.binding.radiobrowser.internal.RadioBrowserHandler;
+import org.openhab.binding.radiobrowser.internal.api.RadioBrowserJson.Country;
+import org.openhab.binding.radiobrowser.internal.api.RadioBrowserJson.Language;
+import org.openhab.binding.radiobrowser.internal.api.RadioBrowserJson.State;
+import org.openhab.binding.radiobrowser.internal.api.RadioBrowserJson.Station;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.types.StateOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link RadioBrowserApi} Handles all http calls to and from the Radio Stations API.
+ *
+ * @author Matthew Skinner - Initial contribution
+ */
+@NonNullByDefault
+public class RadioBrowserApi {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final RadioBrowserHandler handler;
+    private final HttpClient httpClient;
+    private final Gson gson = new Gson();
+    private String server = "";
+    private String language = "";
+    private String countryCode = "";
+    private String state = "";
+    private String genre = "";
+    private Map<String, Station> stationMap = new HashMap<>();
+    public Map<String, Country> countryMap = new HashMap<>();
+
+    public RadioBrowserApi(RadioBrowserHandler handler, HttpClient httpClient) {
+        this.handler = handler;
+        this.httpClient = httpClient;
+    }
+
+    private String sendGetRequest(String url) throws ApiException {
+        Request request;
+        String errorReason = "";
+        request = httpClient.newRequest("http://" + server + url);
```

Comment:
Is this available in https ? If so, please adjust.

Reply:
It was set to https and reverted. It l.ooks liek the API is available in https, anything i'm missing?

Reply:
Yes my time, leaving this open to look at it when I have the time. I have not done https with Jetty before so do not know what needs to change without looking deeper into it. Simply changing the one line to https meant the server gave back 404 errors, guessing it needs extra code to setup the certificate.

Reply:
Got it working via HTTPS